// Code generated by private/model/cli/gen-api/main.go. DO NOT EDIT.

package signer

import (
	"github.com/aws/aws-sdk-go/private/protocol"
)

const (

	// ErrCodeAccessDeniedException for service response error code
	// "AccessDeniedException".
	//
	// You do not have sufficient access to perform this action.
	ErrCodeAccessDeniedException = "AccessDeniedException"

	// ErrCodeBadRequestException for service response error code
	// "BadRequestException".
	//
	// The request contains invalid parameters for the ARN or tags. This exception
	// also occurs when you call a tagging API on a cancelled signing profile.
	ErrCodeBadRequestException = "BadRequestException"

	// ErrCodeInternalServiceErrorException for service response error code
	// "InternalServiceErrorException".
	//
	// An internal error occurred.
	ErrCodeInternalServiceErrorException = "InternalServiceErrorException"

	// ErrCodeNotFoundException for service response error code
	// "NotFoundException".
	//
	// The signing profile was not found.
	ErrCodeNotFoundException = "NotFoundException"

	// ErrCodeResourceNotFoundException for service response error code
	// "ResourceNotFoundException".
	//
	// A specified resource could not be found.
	ErrCodeResourceNotFoundException = "ResourceNotFoundException"

	// ErrCodeThrottlingException for service response error code
	// "ThrottlingException".
	//
	// The signing job has been throttled.
	ErrCodeThrottlingException = "ThrottlingException"

	// ErrCodeValidationException for service response error code
	// "ValidationException".
	//
	// You signing certificate could not be validated.
	ErrCodeValidationException = "ValidationException"
)

var exceptionFromCode = map[string]func(protocol.ResponseMetadata) error{
	"AccessDeniedException":         newErrorAccessDeniedException,
	"BadRequestException":           newErrorBadRequestException,
	"InternalServiceErrorException": newErrorInternalServiceErrorException,
	"NotFoundException":             newErrorNotFoundException,
	"ResourceNotFoundException":     newErrorResourceNotFoundException,
	"ThrottlingException":           newErrorThrottlingException,
	"ValidationException":           newErrorValidationException,
}
